components/client/src/main/java/com/hotels/styx/client/netty/connectionpool/NettyToStyxResponsePropagator.java

```diff
@@ -156,7 +157,17 @@ protected void channelRead0(ChannelHandlerContext ctx, Object msg) throws Except
     @Override
     public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exception {
         if (evt instanceof IdleStateEvent) {
-            getContentProducer(ctx).tearDownResources("idle timeout");
+            FlowControllingHttpContentProducer producer = getContentProducer(ctx);
+            getContentProducer(ctx).tearDownResources(
+                new ResponseTimeoutException(
```

Comment:
A strictly correct behaviour is to use a `ResponseTimeoutException` whenever the HTTP headers have yet not been received, and a `ContentTimeoutException` otherwise. We could use a presence of `producer` to determine this.

But this retains the existing behaviour, and therefore we can do this as a separate increment. Could you raise a GH issue for this please?

Reply:
this is fixed in this branch

Reply:
will do